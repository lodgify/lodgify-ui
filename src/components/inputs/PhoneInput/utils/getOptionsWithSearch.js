/**
 * @param  {Object[]} options
 * @param  {string}   searchValue
 * @return {Object[]}
 */
export const getOptionsWithSearch = (options, searchValue) => {
  const regExp = new RegExp(`^${searchValue.replace('+', '\\+')}`, 'i');

  return options.filter(
    ({ name, value }) => regExp.test(name) || regExp.test(value)
  );
};
