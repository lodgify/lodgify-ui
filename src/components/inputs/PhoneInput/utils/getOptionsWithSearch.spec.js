import { getOptionsWithSearch } from './getOptionsWithSearch';

describe('getOptionsWithSearch', () => {
  it('should return the right array', () => {
    const options = [
      { name: 'AA', value: 'some value' },
      { name: 'AB', value: 'some value' },
      { name: 'CC', value: 'abkhazia' },
      { name: 'DD', value: 'absent minded' },
      { name: 'EE', value: 'some value' },
    ];

    const actual = getOptionsWithSearch(options, 'ab');

    expect(actual).toMatchSnapshot();
  });
});
