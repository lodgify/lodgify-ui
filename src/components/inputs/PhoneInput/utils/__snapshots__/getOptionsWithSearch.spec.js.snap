// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOptionsWithSearch should return the right array 1`] = `
Array [
  Object {
    "name": "AB",
    "value": "some value",
  },
  Object {
    "name": "CC",
    "value": "abkhazia",
  },
  Object {
    "name": "DD",
    "value": "absent minded",
  },
]
`;
